ecs-update/ecs-update.go

```diff
@@ -804,15 +872,27 @@ func (e *packageEditor) modifyBuildManifest() error {
 	if err != nil {
 		return err
 	}
-	err = yamlEditString(f, "$.dependencies.ecs.reference",
-		e.config.BuildManifest.ECSReference, token.DoubleQuoteType)
-	if err != nil {
-		return err
+
+	if e.config.BuildManifest.ECSReference != "" {
+		err = yamlEditString(f, "$.dependencies.ecs.reference",
+			e.config.BuildManifest.ECSReference, token.DoubleQuoteType)
+		if err != nil {
+			return err
+		}
+		e.result.BuildManifest.Changed = true
+		e.result.BuildManifest.ECSReferenceOld = e.pkg.Build.Dependencies.ECS.Reference
+		e.result.BuildManifest.ECSReferenceNew = e.config.BuildManifest.ECSReference
 	}
 
-	e.result.BuildManifest.Changed = true
-	e.result.BuildManifest.ECSReferenceOld = e.pkg.Build.Dependencies.ECS.Reference
-	e.result.BuildManifest.ECSReferenceNew = e.config.BuildManifest.ECSReference
+	if e.config.BuildManifest.ECSImportMappings != nil && !*e.config.BuildManifest.ECSImportMappings {
+		err = yamlDeleteNodeFromMap(f, "$.dependencies.ecs.import_mappings")
+		if err != nil {
+			return err
+		}
+		e.result.BuildManifest.Changed = true
```

Comment:
I would rename `Changed` to `ECSReferenceChanged`. And then create a separate `ECSImportMappingsChanged`. This way the changelog and commit messages can reflect specifically what changed.